src/Fixer/Basic/BracesFixer.php

```diff
@@ -846,6 +846,11 @@ private function ensureWhitespaceAtIndexAndIndentMultilineComment(Tokens $tokens
 
         $nextToken = $tokens[$nextTokenIndex];
         if ($nextToken->isComment()) {
+            $previousToken = $tokens[$nextTokenIndex - 1];
+            // do not indent comments starting at the beginning of line - non-indented comments should not change indentation
+            if ($previousToken->isWhitespace() && 1 === preg_match('/\R$/', $previousToken->getContent())) {
+                return;
```

Comment:
Sorry, but I don't have enough time and knowledge to refactor this fixer. I also need to fix #2986 in 2.2.x line and I highly doubt that such decoupling will be merged into this line.
Can't we just agree that fixing indentation should not be directly connected with fixing braces position? And then consider this PR as reducing damages of bad design of this fixer? Or at least introduce config option to disable indentation fixing? I really need fixer for braces, but don't care about fixing of indentation (especially that at this point it is far from perfect).